### PR TITLE
docs: add parser-risk plan requirements

### PR DESCRIPTION
**Pull request by lhpaul**
## Summary
- add a parser-risk conditional block to plan protocol Step 3 with deterministic classification signals
- add template and review-contract checks for edge-case enumeration, unit-test mapping, and suppression semantics
- align tech-lead agent guidance and add an Unreleased changelog entry for issue #201

## Testing
- npx markdownlint-cli2 \"docs/ai/development-workflow/protocols/02-generate-implementation-plan-protocol.md\" \"docs/ai/development-workflow/templates/implementation-plan-template.md\" \"REVIEW.md\" \"CHANGELOG.md\"

Made with [Cursor](https://cursor.com)

## Release Notes

* **Documentation**
  * Enhanced implementation plan requirements for features involving parser/regex/structured-text scanning with deterministic risk classification
  * Added mandatory subsections for edge-case enumeration and unit test mapping when parser-risk applies
  * New requirements for documenting suppression semantics (directives, placement, multi-suppression behavior) when supported
  * Updated plan review validation rules to enforce parser-risk quality standards

### CODE REVIEW

**Comment by lhpaul** [comment]
### Step 7a Internal Review Gate Summary

**Effective reviewer set**: claude, codex
**Skipped reviewers**: none
**Verdict**: APPROVED

All configured internal reviewers are reachable from the current runner and the implementation review gate is clean.

File: docs/ai/development-workflow/templates/implementation-plan-template.md [hunk 1]
```diff
@@ -70,6 +70,12 @@
 
 **Regression suite**: If the repository has an automated regression test suite, include a checklist item in the relevant layer for a new regression spec that covers the smoke test runbook scenarios above. Omit this if no regression suite exists in the repository.
 
+### Parser-risk addendum (include only when Step 3 classifier applies)
+
+- **Edge-case enumeration**: List concrete parser/scanner inputs that cover boundary variants, negative lookalikes, multi-match lines, and nested/overlapping patterns when relevant.
+- **Unit test mapping**: Name a unit test file and map at least one automated unit test per enumerated edge case.
+- **Suppression semantics (if applicable)**: Name recognized suppression directives, allowed placement, and behavior when multiple suppressions appear on one line.
```
[the remark this answers is not in the document]
Comment (lhpaul) [reply docs/ai/development-workflow/templates/implementation-plan-template.md]: Fixed in commit `980c08a`.

File: REVIEW.md [hunk 2]
```diff
@@ -89,6 +89,10 @@ Check:
 - Steps are specific enough to execute without guessing
 - Ordering is feasible and dependencies are explicit
 - When pattern-based completeness applies, enumerated counts/paths are validated against the plan's Verification Log commands and outputs
+- Parser-risk completeness (when protocol `02-generate-implementation-plan-protocol.md` Step 3 parser-risk signals apply):
+  - Edge-case enumeration is present and concrete
+  - A unit test file is named with at least one automated test mapped per enumerated case
+  - Suppression semantics subsection is present when suppressions are part of the proposed feature
```
[the remark this answers is not in the document]
Comment (lhpaul) [reply REVIEW.md]: Fixed in commit `980c08a`.

**Comment by lhpaul** [comment]
### Automated Fix: commit `980c08a`

Addressed **2** finding(s) from cycle 1:

| # | Platform | File | Description |
|---|----------|------|-------------|
| 1 | coderabbit | `docs/ai/development-workflow/templates/implementation-plan-template.md:77` | Added normative-spec flexibility requirement to parser-risk edge-case enumeration. |
| 2 | coderabbit | `REVIEW.md:95` | Expanded suppression-semantics checklist to require directives, placement, and multi-suppression behavior. |

**Comment by lhpaul** [comment]
### Step 7a Internal Review Gate Summary

**Effective reviewer set**: claude, codex
**Skipped reviewers**: none
**Verdict**: APPROVED

Re-ran Step 7a after automated-review fixes; internal gate remains clean.

File: docs/ai/development-workflow/templates/implementation-plan-template.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (lhpaul) [reply docs/ai/development-workflow/templates/implementation-plan-template.md]: Fixed in commit `cd48e0a`.

File: REVIEW.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (lhpaul) [reply REVIEW.md]: Fixed in commit `cd48e0a`.

[the remark this answers is not in the document]
Comment (lhpaul) [reply REVIEW.md]: Fixed in commit `cd48e0a`.

File: docs/ai/development-workflow/templates/implementation-plan-template.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (lhpaul) [reply docs/ai/development-workflow/templates/implementation-plan-template.md]: Fixed in commit `cd48e0a`.

File: .claude/agents/tech-lead.md [hunk 3]
```diff
@@ -11,4 +11,6 @@ Follow the implementation plan generation protocol exactly as defined in:
 
 That document is the single source of truth for this stage. Always read the approved spec (or the work item brief for Refactor items) and relevant codebase sections before proposing an approach. Once ambiguity is resolved, continue through reviewer gate, PR creation, and PR readiness unless the protocol requires human input.
 
+Before finalizing Step 3, classify parser-risk using the deterministic signals in protocol 02 (tooling-path parser/lint changes, parser/scanner-oriented module naming, or explicit regex/structured-text scanning behavior). When parser-risk applies, include the mandatory edge-case enumeration and unit-test mapping subsections before deep Layer-by-Layer walkthroughs. If suppressions are part of the feature, include suppression semantics (recognized directives, placement, and multi-suppression behavior).
```
[the remark this answers is not in the document]
Comment (lhpaul) [reply .claude/agents/tech-lead.md:14]: Fixed in commit `cd48e0a`.

**Comment by lhpaul** [comment]
### Automated Fix: commit `cd48e0a`

Addressed **1** finding(s) from cycle 2:

| # | Platform | File | Description |
|---|----------|------|-------------|
| 1 | devin | `.codex/skills/workflow-plan-writer/SKILL.md:15` | Added parser-risk Step 3 reminder to keep Codex plan-writer guidance aligned with Claude/Cursor entrypoints. |

**Comment by lhpaul** [comment]
### Step 7a Internal Review Gate Summary

**Effective reviewer set**: claude, codex
**Skipped reviewers**: none
**Verdict**: APPROVED

Re-ran Step 7a after Devin-requested fixes; internal gate remains clean.

**Comment by lhpaul** [comment]
### Automated Reviewer Loop Summary

**Result:** clean
**Cycles:** 2 / 10
**Platforms:** devin, coderabbit

| # | Platform | File | Line | Description | Status | Resolved In |
|---|----------|------|------|-------------|--------|-------------|
| 1 | coderabbit | `docs/ai/development-workflow/templates/implementation-plan-template.md` | 77 | Missing normative-spec flexibility category in parser-risk edge-case template bullet. | Resolved | `980c08a` |
| 2 | coderabbit | `REVIEW.md` | 95 | Suppression semantics check needed explicit directive/placement/multi-suppression criteria. | Resolved | `980c08a` |
| 3 | devin | `.codex/skills/workflow-plan-writer/SKILL.md` | 15 | Codex plan-writer entrypoint missing parser-risk Step 3 reinforcement. | Resolved | `cd48e0a` |

**Resolved:** 3 / 3 findings
